src/wrap-store/shallowDiff.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * Returns a new Object containing only the fields in `new` that differ from `old`
+ *
+ * @param {Object} old
+ * @param {Object} new
+ * @return {Object{updated, removed}} An object containing the updated and removed
+ *   fields. Updated is an object containing any fields added or changed. Removed
+ *   is an array of keys that exist in the old object but are missing in the new object.
+ */
+export default function shallowDiff(oldObj, newObj) {
+  const updated = {};
+  const removed = [];
+
+  Object.keys(newObj).forEach((key) => {
```

Comment:
We should probably condense things to one loop (as the state could potentially have many keys - also a little cleaner).

Reply:
Without modifying objects, I'm not sure how I could optimize this loop. I have to loop over the new object to get what has been updated or added, and then over the old object to see what was removed.